### PR TITLE
fix: fail when mantra license error is seen

### PR DESCRIPTION
**Pull request by epmog**
### What was the problem/requirement? (What/Why)

Houdini jobs weren't failing properly if a license error occurred when launching hython:

```
INFO: Running command: hython "/Users/morgane/Library/Application Support/hatch/env/virtual/deadline-cloud-for-houdini/Bp-p5MaR/deadline-cloud-for-houdini/lib/python3.10/site-packages/deadline/houdini_adaptor/HoudiniClient/houdini_client.py"
STDERR: No licenses could be found to run this application.
STDERR: 	Please check for a valid license server host
STDERR: Details: http://localhost:1715: No server found running. Please start the server before retrying.
```

### What was the solution? (How)

capture the license error and fail the job

### What is the impact of this change?

Jobs properly fail!

### How was this change tested?

Ran hython locally without a license connection and added a unit test

```
$ houdini-openjd run --init-data "{'scene_file': 'C:/asdfasdf', 'render_node': 'mantra1', 'version': '19.5'}"
INFO: Applying user-level configuration: /Users/morgane/.openjd/adaptors/runtime/configuration.json
INFO: Applying user-level configuration: /Users/morgane/.openjd/adaptors/HoudiniAdaptor/HoudiniAdaptor.json
INFO: Loading 'init_data' schema from /Users/morgane/Library/Application Support/hatch/env/virtual/deadline-cloud-for-houdini/Bp-p5MaR/deadline-cloud-for-houdini/lib/python3.10/site-packages/deadline/houdini_adaptor/HoudiniAdaptor/schemas/init_data.schema.json
INFO: Loading 'run_data' schema from /Users/morgane/Library/Application Support/hatch/env/virtual/deadline-cloud-for-houdini/Bp-p5MaR/deadline-cloud-for-houdini/lib/python3.10/site-packages/deadline/houdini_adaptor/HoudiniAdaptor/schemas/run_data.schema.json
openjd_progress: 0
openjd_status: Initializing Houdini
INFO: Setting HOUDINI_PATHMAP to: 
INFO: Running command: hython "/Users/morgane/Library/Application Support/hatch/env/virtual/deadline-cloud-for-houdini/Bp-p5MaR/deadline-cloud-for-houdini/lib/python3.10/site-packages/deadline/houdini_adaptor/HoudiniClient/houdini_client.py"
STDERR: No licenses could be found to run this application.
STDERR: 	Please check for a valid license server host
STDERR: Details: http://localhost:1715: No server found running. Please start the server before retrying.
STDERR: 
STDERR: 
openjd_fail: Error encountered while starting adaptor: Houdini encountered a license error: No licenses could be found to run this application.
ERROR: Error running the adaptor: Houdini encountered a license error: No licenses could be found to run this application.
ERROR: Entrypoint failed: Houdini encountered a license error: No licenses could be found to run this application.
```

### Was this change documented?

N/A

### Is this a breaking change?

Nope

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py [hunk 1]
```diff
@@ -200,26 +199,29 @@ def _get_regex_callbacks(self) -> list[RegexCallback]:
         if not self._regex_callbacks:
             callback_list = []
 
-            _houdini_license_error = "RuntimeError: Error encountered when initializing Houdini"
-
             completed_regexes = [re.compile(".*Finished Rendering.*")]
             progress_regexes = [re.compile(".*ALF_PROGRESS ([0-9]+)%.*")]
-            error_regexes = [re.compile(".*Error: .*|.*\\[Error\\].*", re.IGNORECASE)]
+            license_regexes = [
+                # generic runtime error
```
Comment (justinsaws) [inline src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py:205]: Should this be under the license error regex list if it's a generic error?

File: src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py [hunk 2]
```diff
@@ -255,34 +257,26 @@ def _handle_progress(self, match: re.Match) -> None:
...
         Raises:
             RuntimeError: Always raises a runtime error to halt the adaptor.
         """
-        shutil_usage = shutil.disk_usage(os.getcwd())
-        self._exc_info = RuntimeError(
-            f"{match.group(0)}\n"
-            "This error is typically associated with a licensing error"
-            " when using Houdini. Check your licensing configuration.\n"
-            f"Free disc space: {shutil_usage.free//1024//1024}M\n"
-        )
```
Comment (crowecawcaw) [inline src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py:285]: Just curious, why did we report the free disk space when we suspected license errors before?

**Review by crowecawcaw — APPROVED** [review]

File: src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py (hunk 1 above)
Comment (epmog) [reply src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py:205]: I went back and forth on this a few times. It was previously just a license error, so I'm not changing functionality here.

Putting it with the generic errors would then make the "strict error handling" apply to it which I'm not sure we want.

File: src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py (hunk 2 above)
Comment (epmog) [reply src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py:285]: This was due to copy/pasting some hyper-specific maya license handling. They erroneously report licensing errors on the cloud by saying they ran out of disk space. So we did a few attempts at improving it to ensure that the user could easily verify that was not the case.

File: src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py (hunk 1 above)
Comment (justinsaws) [reply src/deadline/houdini_adaptor/HoudiniAdaptor/adaptor.py:205]: Sounds good.

**Review by justinsaws — APPROVED** [review]

**Pull request merged**